lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -357,7 +357,127 @@ namespace Js
+        }
+
+        // If we don't have a tag or it's not a string, use the 'built in tag'.
+        if (isArray)
+        {
+            // 5. If isArray is true, let builtinTag be "Array".
+            return library->CreateStringFromCppLiteral(_u("[object Array]"));
+        }
+
+        JavascriptString* builtInTag = nullptr;
+        switch (type)
+        {
+            // 6. Else if O is an exotic String object, let builtinTag be "String".
+        case TypeIds_String:
+        case TypeIds_StringObject:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object String]"));
+            break;
+
+            // 7. Else if O has an[[ParameterMap]] internal slot, let builtinTag be "Arguments".
+        case TypeIds_Arguments:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object Arguments]"));
+            break;
+
+            // 8. Else if O has a [[Call]] internal method, let builtinTag be "Function".
+        case TypeIds_Function:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object Function]"));
+            break;
+
+            // 9. Else if O has an [[ErrorData]] internal slot, let builtinTag be "Error".
+        case TypeIds_Error:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object Error]"));
+            break;
+
+            // 10. Else if O has a [[BooleanData]] internal slot, let builtinTag be "Boolean".
+        case TypeIds_Boolean:
+        case TypeIds_BooleanObject:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object Boolean]"));
+            break;
+
+            // 11. Else if O has a [[NumberData]] internal slot, let builtinTag be "Number".
+        case TypeIds_Number:
+        case TypeIds_Int64Number:
+        case TypeIds_UInt64Number:
+        case TypeIds_Integer:
+        case TypeIds_NumberObject:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object Number]"));
+            break;
+
+            // 12. Else if O has a [[DateValue]] internal slot, let builtinTag be "Date".
+        case TypeIds_Date:
+        case TypeIds_WinRTDate:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object Date]"));
+            break;
+
+            // 13. Else if O has a [[RegExpMatcher]] internal slot, let builtinTag be "RegExp".
+        case TypeIds_RegEx:
+            builtInTag = library->CreateStringFromCppLiteral(_u("[object RegExp]"));
+            break;
+
+            // 14. Else, let builtinTag be "Object".
+        default:
+            builtInTag = library->GetObjectDisplayString(); // [object Object]
+            break;
+        }
+
+        Assert(builtInTag != nullptr);
+
+        return builtInTag;
+    }
+
+    // This is the previous implementation of @@toStringTag, which is used in some cases even when the feature is disabled.
+    // Leaving this here for now due to the fragility.
+    // TODO(tcare): Remove when @@toStringTag is enabled by default.
```

Comment:
Was it enabled recently? Because my 7/8/16 clone is not showing ES6ToStringTag enabled through FLAGPR_REGOVR_EXP in ConfigFlagsList.h

---
In reply to: [71397520](https://github.com/Microsoft/ChakraCore/pull/1288#discussion_r71397520) [](ancestors = 71397520)